### DISCUSSION TITLE
Java LSP options (Docs)

### DISCUSSION THREAD

**Discussion by spacelatte** [zed-industries/zed#21268] [Feature Requests]
Hello, 

I am currently facing an issue where Java LSP is throwing OutOfMemory (Heap) exception due to a large project.

I was digging [the documentation](https://zed.dev/docs/languages/java) to increase the `Xmx` parameter, I realised 2 things.

1. There are 2 Java extensions, one is just `Java` other one is `JDTLS`. I have both installed and not sure what are the differences.
    1. https://github.com/zed-extensions/java
    2. https://github.com/ABckh/zed-java-eclipse-jdtls
2. The documentation have inconsistent schema in regards to the configuration, some of them may be from the _older_ schema. I did a quick search here and couldn't find a deliberate reference which one is current. 

    1. **1st example:**
    ```
    {
      "lsp": {
        "jdtls": {
          "settings": {
          ...
    ```
    2. **2nd example:**
    ```
    {
      "lsp": {
        "java": {
          "settings": {},
          "initialization_options": {}
        }
      }
    }
    ```
    3. **3rd example:**
    ```
    {
      "lsp": {
        "jdtls": {
          "initialization_options": {
          ...
    ```
    4. **4th example:**
    ```
    {
      "lsp": {
        "java": {
          "settings": {
            "java.jdt.ls.lombokSupport.enabled:": true
          }
        }
      }
    }
    ```

---

To clarify I am looking for following;
- What is the main "key" under `lsp`? (`java` or `jdtls`)
- Where to define "settings"? Directly at the / under top-level, or under `initialization_options`/`settings`?
- If I were going to set `java.jdt.ls.vmargs` to `-Xmx4g`, which one should I do?
    1. `{ lsp: { jdtls: { settings: { "java.jdt.ls.vmargs": "-Xmx4g" } } } }`
    2. `{ lsp: { java:  { settings: { "java.jdt.ls.vmargs": "-Xmx4g" } } } }`
    3. `{ lsp: { jdtls: { initialization_options: { settings: { "java.jdt.ls.vmargs": "-Xmx4g" } } } } }`
    4. `{ lsp: { java:  { initialization_options: { settings: { "java.jdt.ls.vmargs": "-Xmx4g" } } } } }`

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
